### PR TITLE
Docs: update python-api-intro.md

### LINKED ISSUE #4150: Docs: update python-api-intro.md
- I found wrong variable name `partition` in [python-api-intro.md](https://github.com/apache/iceberg/blob/master/docs/versioned/api/python-api-intro.md)
스크린샷 2022-02-17 오후 4 38 24
I think it should be `partition_spec`
Would you please check it?
If you are ok, I wish I could open PR for fixing it by myself.
-  missing configuration `hive.metastore.warehouse.dir` for initializing  HiveTables
After fixed first problem, It still does not work property because of missing configuration key for `hive.metastore.warehouse.dir`

### PR DESCRIPTION
**Pull request by happyprg**
closed #4150

### CODE REVIEW

File: docs/versioned/api/python-api-intro.md [hunk 1]
```diff
@@ -36,7 +36,8 @@ To create a catalog:
 from iceberg.hive import HiveTables
 
 # instantiate Hive Tables
-conf = {"hive.metastore.uris": 'thrift://{hms_host}:{hms_port}'}
+conf = {"hive.metastore.uris": 'thrift://{hms_host}:{hms_port}',
+        "hive.metastore.warehouse.dir": tmpdir}
```
[the remark this answers is not in the document]
Comment (happyprg) [reply docs/versioned/api/python-api-intro.md:40]: > curious to know, why add this line as well ?

@dramaticlly 
Thank you for checking my PR.
I thought that When I setup `hive.metastore.uris` property, Warehouse location is required mandatory.
I checked [test case codes of Iceberg](https://github.com/apache/iceberg/blob/master/python_legacy/tests/hive/test_hive_tables.py#L131) also.
스크린샷 2022-02-18 오전 10 15 00
스크린샷 2022-02-18 오전 10 15 12
스크린샷 2022-02-18 오전 10 15 23

Am I misunderstanding?

[the remark this answers is not in the document]
Comment (dramaticlly) [reply docs/versioned/api/python-api-intro.md:40]: yeah I guess the legacy python API asked to supply the valid hive metastore warehouse directory and fails the code if default to None. But I guess `tmpdir` is not resolved directly, if needs proper setup I guess need to use the python module https://docs.python.org/3/library/tempfile.html#module-tempfile.

[the remark this answers is not in the document]
Comment (happyprg) [reply docs/versioned/api/python-api-intro.md]: I changed text `tmpdir` to `{tmpdir}`. [following current convention(replacement text)](https://github.com/apache/iceberg/pull/4151/files#diff-e2fa3022513566d7037734d0c167b142be6f86df26aa59fe5548eff6182dd15bL39) 

how about this?

[the remark this answers is not in the document]
Comment (dramaticlly) [reply docs/versioned/api/python-api-intro.md]: but I think PMC is needed to merge your code. CC @jun-he

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @happyprg!

**Comment by rdblue** [comment]
Sorry, I thought I already had. Thanks for pinging me.

**Pull request merged**